### PR TITLE
Use github workflow for Codecov.io report

### PR DESCRIPTION
**Pull request by ephphatha**
...not entirely sure how to actually trigger this without merging it...

### CODE REVIEW

File: .github/workflows/Linux_x86_64_test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install dependencies
+        run: |
+          apt-get update -y
+          apt-get install -y cmake curl g++ git lcov libgtest-dev libgmock-dev libfmt-dev libsdl2-dev libsodium-dev libpng-dev libbz2-dev
```
Comment (ephphatha) [inline .github/workflows/Linux_x86_64_test.yml:22]:
```suggestion
          sudo apt-get update -y
          sudo apt-get install -y cmake curl g++ git lcov libgtest-dev libgmock-dev libfmt-dev libsdl2-dev libsodium-dev libpng-dev libbz2-dev
```

File: .github/workflows/Linux_x86_64_test.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    types: [ opened, synchronize ]
+
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
```
Comment (ephphatha) [inline .github/workflows/Linux_x86_64_test.yml:17]:
```suggestion
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
```

File: .github/workflows/Linux_x86_64_test.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y cmake curl g++ git lcov libgtest-dev libgmock-dev libfmt-dev libsdl2-dev libsodium-dev libpng-dev libbz2-dev
+      - name: Build tests
+        run: |
+          cmake -S. -Bbuild -DENABLE_CODECOVERAGE=ON
+          cmake --build build -j 2
```
Comment (ephphatha) [inline .github/workflows/Linux_x86_64_test.yml:26]:
```suggestion
          cmake --build build -j $(nproc)
```

**Review by ephphatha — COMMENTED** [review]
need to fetch tags

**Review by AJenbo — APPROVED** [review]
Thanks

**Comment by AJenbo** [comment]
I would have expected to see a report on codecov, but it appears to be giving a reference error instead... Oh well let's try it.

**Pull request merged**